notebooks/AsymmetricErrors.ipynb

```diff
@@ -0,0 +1,577 @@
+      "text/plain": [
+       "<Figure size 640x480 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "ui.reg_proj(pl.gamma, pl.ampl, nloop=[101, 101])\n",
+    "add_conf_gamma()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2a9b27dd-afb9-44c0-83cf-8564d9ac52ba",
+   "metadata": {},
+   "source": [
+    "So, **all** the above used the average error values when calculating statistic values.\n",
+    "\n",
+    "The trick to using the asymmetric errors is to use the `resample_data` call to simulate drawing the data using the asymmetric errors and fitting the resulting simulated data. In the following we explicitly ask for 1000 iterations (although that is the default value) and fix the random-number seed. The call displays information on the distribution of the fit parameters:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "dfe66669-a317-45f0-9077-574183bbf9ba",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "pl.gamma : avg = -0.41801959511066183 , std = 0.1984493399280746\n",
+      "pl.ampl : avg = 163.0552687577236 , std = 94.50832743946526\n"
+     ]
+    }
+   ],
+   "source": [
+    "result = ui.resample_data(niter=1000, seed=123)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc6f7ad8-b5d2-466e-8315-e687722b9f7e",
+   "metadata": {},
+   "source": [
+    "The routine returns a dictionary with arrays as values; we can check what the keys are with:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "f5577f8f-ce02-4d24-9d64-4bb1878dd028",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "dict_keys(['samples', 'statistic', 'pl.gamma', 'pl.ampl'])\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(result.keys())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2470998c-2cbd-47de-86fa-9309e38d5d8c",
+   "metadata": {},
+   "source": [
+    "A trace of the statistic value can be created. Unfortunately the documentation is unclear on what this statistic is, but it **appears** to be the least-squares difference between the simulated data and the best-fit model:"
```

Comment:
`resample_data` help file describes the output:

> Returns
> :
> sampled – The keys are statistic, which contains the best-fit statistic value for each iteration, samples, which contains the resampled data used in the fits as a niter by ndata array, and the free parameters in the fit, containing a NumPy array containing the fit parameter for each iteration (of size niter).
>

Reply:
I still claim the documentation is incomplete because it just says "statistic", not what sort of statistic. I believe, from looking at the code, that it is the least-square statistic (you cna see, from the values, that it doesn't match the chi2 values from the fit).

Reply:
the fit statistics - so if you fit with leastsq it should be leastsq, if with chi2 then it should be chi2.

my old notebook used leastsq.

Reply:
This was why I said in the intro that all I was doing was a minimal fix as there appears to be a number of issues we could have done better when this was originally written. The "fit to the resampled data" is done in

https://github.com/sherpa/sherpa/blob/45d2379c67e7c49b6ca6c79ffe173075ddf1d5fa/sherpa/sim/__init__.py#L970-L992

and you can see that the statistic and optimization method are hard-coded (line 975) to `LeastSq` and `LevMar`. Ideally these could be set by the user in some manner, but I don't really want to do that in this PR!

Reply:
I've created #2093 to track this particular issue

Reply:
So, I've basically said that it's the least-squares statistic and it currently can't be changed.